### ISSUE TITLE
ConditionalNodeDecision not invoked for every field

### ISSUE THREAD

**Issue by dfa1** [graphql-java/graphql-java#3409]
**Describe the bug**
I'm trying to use ConditionalNodeDecision but it not invoked for every field, just the top level one.

**To Reproduce**

```java
final ExecutionInput executionInput = ExecutionInput.newExecutionInput()
                .query(request.query())
                .operationName(request.operationName())
                .variables(request.variables())
                .graphQLContext(builder ->
.graphQLContext(builder ->
                        builder
.put(ConditionalNodeDecision.class, new Custom()) <-- 
```

but then in graphql.execution.conditional.ConditionalNodes#shouldInclude:
```java
        // if they have declared a decision callback, then we will use it but we expect this to be mostly
        // empty and hence the cost is a map lookup.
        if (graphQLContext != null) {
            ConditionalNodeDecision conditionalDecision = graphQLContext.get(ConditionalNodeDecision.class);
```

only the first call has the context, subsequent calls have empty context. 

The first call is originating in:
![image](https://github.com/graphql-java/graphql-java/assets/5798698/686ee057-c69b-4994-8071-15074f2a3e15)

and as you can see the context is defined.
The next call is originating in `getImmediateFields`:
![image](https://github.com/graphql-java/graphql-java/assets/5798698/7bc7905e-1cbd-4e20-aa26-ab8688320d20)
and at this point the context is empty.

Graphql-java version is 21.2 on JDK 17.

**Comment by dondonz** [comment]
Thanks for reporting the bug! I was able to reproduce the problem and will take a closer look

**Comment by dondonz** [comment]
Thanks again for the bug report! The fix is in #3411, the fix is to propagate the GraphQLContext when completing values as well.

**Comment by dfa1** [comment]
@dondonz thanks for the quick fix. Would it be possible to merge to master? I can do some tests on my side as well :)

**Comment by dondonz** [comment]
Yes it'll be merged soon, just waiting on a review

This branch is very close to the current HEAD of master

**Comment by dondonz** [comment]
Fixed by #3411
